### PR TITLE
normalized apps

### PR DESCRIPTION
**Pull request by Kelvinrr**
## Description

This makes it so all app functions are speaking the same language. The purpose of this being a brain dead layer on top of apps to move faster on app conversions and allow for a more laser-guided approach later. 

## Motivation and Context

We had 3 different app-to-function implementations. We need to pick one to move forward with reducing the testing burden on ISIS. 

## How Has This Been Tested?

Ran all the tests on converted apps locally. 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

**Comment by Kelvinrr** [comment]
If this gets merged are we good to remove spiceinit app tests and their data? The spiceinit tests seem to cover all current spiceinit tests.

**Comment by Kelvinrr** [comment]
I appended the Cube stuff onto here since it was a relatively minor change and depended on this PR.

**Review by scsides — COMMENTED** [review]
Looks mostly innocuous, has this been tested with the web service?

File: isis/tests/Fixtures.cpp [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    }
+  }
+
+  void TestCube::createCube(Pvl &label, nlohmann::json &isd) {
+    createCube(label);
+    testCube.attachSpiceFromIsd(isd);
+    
+    // close cube to flush and write tables
+    testCube.close();
+    testCube.open(tempFile.fileName()+".cub");
```
Comment (Kelvinrr) [inline isis/tests/Fixtures.cpp:35]: We might run into more issues like this where we need to close the cubes and re-open after changes. Seems cubes were not designed to live prolonged lives as in-memory objects.

File: isis/src/base/apps/spiceinit/spiceinit.cpp [hunk 2]
```diff
@@ -20,75 +18,51 @@
...
+  void requestSpice(Cube *icube, UserInterface &ui, Pvl *log, Pvl &labels, QString missionName);
 
-  /**
-   * Spiceinit a cube in an Application
-   *
-   * @param ui The Application UI
-   * @param(out) log The Pvl that attempted kernel sets will be logged to
-   */
-  void spiceinit(UserInterface &ui, Pvl *log) {
+  void spiceinit(UserInterface &ui, Pvl *log) {
```
Comment (kaitlyndlee) [inline isis/src/base/apps/spiceinit/spiceinit.cpp:46]: What happened to the method documentation?

File: isis/src/base/apps/spiceinit/spiceinit.cpp [hunk 3]
```diff
@@ -20,75 +18,51 @@
...
-  /**
-   * Spiceinit a Cube
-   *
-   * @param cube The Cube to spiceinit
-   * @param options The options for how the cube should be spiceinit'd
-   * @param(out) log The Pvl that attempted kernel sets will be logged to
-   */
-  void spiceinit(Cube *icube, const spiceinitOptions &options, Pvl *log) {
+  
+  void spiceinit(Cube *icube, UserInterface &ui, Pvl *log) {
```
Comment (kaitlyndlee) [inline isis/src/base/apps/spiceinit/spiceinit.cpp:64]: Same as above.

File: isis/src/base/apps/spiceinit/spiceinit.cpp [hunk 4]
```diff
@@ -276,125 +252,31 @@ namespace Isis {
...
-    * @param iak The instrument addendum kernels
-    * @param dem The digital elevation model
-    * @param exk The extra kernels
-    *
-    * @return If a camera model was successfully created
-   */
-  bool tryKernels(Cube *icube, Process &p,
-                  const spiceinitOptions &options,
-                  Pvl *log,
+  /**
```
Comment (kaitlyndlee) [inline isis/src/base/apps/spiceinit/spiceinit.cpp]: Same as above, why remove most of the documentation?

File: isis/src/base/apps/spiceinit/spiceinit.cpp [hunk 5]
```diff
@@ -673,67 +555,37 @@ namespace Isis {
...
-   * @param missionName The NAIF name of the mission the Cube is from
-   * @param options The spiceinit options
-   * @param log The Application log
-   */
-  void requestSpice(Cube *icube,
-                    Pvl &labels,
-                    QString missionName,
-                    const spiceinitOptions &options,
-                    Pvl *log) {
+  void requestSpice(Cube *icube, UserInterface &ui, Pvl *log, Pvl &labels, QString missionName) {
```
Comment (kaitlyndlee) [inline isis/src/base/apps/spiceinit/spiceinit.cpp:602]: Same as above.

File: isis/src/base/apps/stats/stats.cpp [hunk 6]
```diff
@@ -16,19 +16,24 @@ using namespace Isis;
 
 namespace Isis {
 
+  
+  void stats(UserInterface &ui) {
```
Comment (kaitlyndlee) [inline isis/src/base/apps/stats/stats.cpp:25]: Add documentation

File: isis/src/base/objs/Cube/Cube.cpp [hunk 7]
```diff
@@ -1170,6 +1171,72 @@ namespace Isis {
...
+    PvlKeyword ckKeyword("InstrumentPointing");
+    PvlKeyword ikKeyword("Instrument");
+    PvlKeyword sclkKeyword("SpacecraftClock");
+    PvlKeyword spkKeyword("InstrumentPosition");
+    PvlKeyword iakKeyword("InstrumentAddendum");
+    PvlKeyword demKeyword("ShapeModel");
+    PvlKeyword exkKeyword("Extra");  
+    
+    Spice spice(*this->label(), isd); 
+    std::cout << "done" << std::endl;
```
Comment (kaitlyndlee) [inline isis/src/base/objs/Cube/Cube.cpp]: Forgot to remove cout line?

**Comment by Kelvinrr** [comment]
@scsides doesn't seem like the spiceinit app tests tested the web service. Jesse originally converted the tests that were there for spiceinit and added some for kerneldb and I'm assuming the web service gets tested separately in a unit test.

File: isis/src/base/apps/spiceinit/spiceinit.cpp (hunk 4 above)
Comment (Kelvinrr) [reply isis/src/base/apps/spiceinit/spiceinit.cpp]: There were a lot of changes in there originally that needed to be reverted so it was easier to revert the file and than manually bring things back in. I guess I forgot the documentation strings.

File: isis/tests/spiceinitTests.cpp [hunk 8]
```diff
@@ -11,43 +9,17 @@
...
 
-    for (auto grpIt = cubeLabel.beginGroup(); grpIt!= cubeLabel.endGroup(); grpIt++) {
-      testCube.putGroup(*grpIt);
-    }
-  }
-};
-
-
-TEST_F(spiceinitTestCube, PredictAndReconCK) {
+TEST_F(TestCube, spiceinit_predict_and_recon_ck) {
```
Comment (kaitlyndlee) [inline isis/tests/spiceinitTests.cpp]: Not a big deal, just wondering if we are swapping to using underscores instead of camel case for test names.

**Comment by jessemapel** [comment]
@Kelvinrr The spice web service gets tests as a separate app https://github.com/USGS-Astrogeology/ISIS3/tree/dev/isis/src/base/apps/spiceserver/tsts

File: isis/tests/spiceinitTests.cpp (hunk 8 above)
Comment (Kelvinrr) [reply isis/tests/spiceinitTests.cpp]: This should be back to camel case I think.

**Review by kaitlyndlee — APPROVED** [review]

**Pull request merged**